core/src/main/scala/org/apache/spark/rdd/CartesianRDD.scala

```diff
@@ -57,7 +58,12 @@ class CartesianRDD[T: ClassTag, U: ClassTag](
 
   override def getPartitions: Array[Partition] = {
     // create the cross product split
-    val array = new Array[Partition](rdd1.partitions.length * rdd2.partitions.length)
+    val numPartitionsInRdd1 = rdd1.partitions.length
+    val partitionNum: Long = numPartitionsInRdd1.toLong * numPartitionsInRdd2.toLong
+    if (partitionNum > Int.MaxValue) {
+      throw SparkCoreErrors.cartesianPartitionNumOverflow(numPartitionsInRdd1, numPartitionsInRdd2)
```

Comment:
I prefer to reuse COLLECTION_SIZE_LIMIT_EXCEEDED

Reply:
@yaooqinn Thank you for your suggestion, let me change it.